lib/slax_web/websocket_listener.ex

```diff
@@ -0,0 +1,77 @@
+defmodule SlaxWeb.WebsocketListener do
+  use GenServer
+
+  alias Slax.Http
+  alias SlaxWeb.Issue
+  alias SlaxWeb.Poker
+
+  defp config() do
+    Application.get_env(:slax, Slax.Slack)
+  end
+
+  defp app_token() do
+    config()[:app_token]
+  end
+
+  def init(_) do
+    {:ok, %{body: %{"url" => "wss://wss-primary.slack.com" <> path}}} =
+      Http.post(
+        "https://slack.com/api/apps.connections.open",
+        "",
+        "Content-Type": "application/x-www-form-urlencoded",
+        Authorization: "Bearer #{app_token()}"
+      )
+
+    {:ok, pid} =
+      :gun.open(:binary.bin_to_list("wss-primary.slack.com"), 443, %{
+        transport: :tls,
+        protocols: [:http],
+        tls_opts: [verify: :verify_none, log_level: :debug]
+      })
+
+    {:ok, :http} = :gun.await_up(pid, 10_000)
+    stream = :gun.ws_upgrade(pid, path)
+    {:upgrade, [<<"websocket">>], _headers} = :gun.await(pid, stream, 10_000)
+
+    {:ok, nil}
+  end
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, nil)
+  end
+
+  def handle_info({_, pid, stream_ref, {:text, event}}, socket) do
+    {:ok, decoded_event} = Jason.decode(event)
```

Comment:
do we want to handle the error case here?

Reply:
By handle the error case do you mean using `decode!` instead?

Reply:
yeah or pattern matching on the result of `decode` so that you can handle the case when it returns something other than {:ok,

Reply:
Yeah I went ahead and changed it to `decode!`

Reply:
I am wondering if it explodes if it will crash the genserver and if that's the behavior we want? or if we want to use case decode/1 do and then on the error case we log an error message and {noreply, socket} or if it matters?

Reply:
Yeah I think it'll crash. That's probably OK for now b/c since it's a child of the top level supervisor, which should restart it due to `strategy: :one_for_one`